zulipterminal/ui_tools/boxes.py

```diff
@@ -1690,23 +1700,45 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
-                time_since_msg_sent = time() - self.message["timestamp"]
-                edit_time_limit = self.model.initial_data[
-                    "realm_message_content_edit_limit_seconds"
-                ]
-                if time_since_msg_sent >= edit_time_limit:
-                    if self.message["type"] == "private":
-                        self.model.controller.report_error(
-                            " Time Limit for editing the message has been exceeded."
-                        )
-                        return key
-                    elif self.message["type"] == "stream":
+                if self.message["sender_id"] == self.model.user_id:
+                    time_since_msg_sent = time() - self.message["timestamp"]
+                    edit_time_limit = self.model.initial_data[
+                        "realm_message_content_edit_limit_seconds"
+                    ]
+                    # Don't allow editing message body if time-limit exceeded.
+                    if time_since_msg_sent >= edit_time_limit:
+                        if self.message["type"] == "private":
+                            self.model.controller.report_error(
+                                " Time Limit for editing the message has been exceeded."
+                            )
+                            return key
+                        elif self.message["type"] == "stream":
+                            self.model.controller.report_warning(
+                                " Only topic editing allowed."
+                                " Time Limit for editing the message body"
+                                " has been exceeded."
+                            )
+                            msg_body_edit_enabled = False
+                elif self.message["type"] == "stream":
+                    # Allow editing topic if the message has "(no topic)"
+                    if self.message["subject"] == "(no topic)":
                         self.model.controller.report_warning(
-                            " Only topic editing allowed."
-                            " Time Limit for editing the message body has"
-                            " been exceeded.",
+                            " Only topic editing is allowed."
                         )
                         msg_body_edit_enabled = False
+                    # FIXME: Is this else necessary, we already handle this at the top?
+                    else:
+                        self.model.controller.report_error(
+                            " You can't edit messages sent by other users that"
+                            " already have a topic."
+                        )
+                        return key
+                else:
+                    # The remaining case is of a private message not belonging to user.
+                    self.model.controller.report_error(
+                        " You can't edit private messages sent by other users."
```

Comment:
You're not testing this - is it already covered?

Reply:
Actually, because of the equivalent conditional at the top, this case was already handled by that and returned so we never reached here.
Removing the top block as I suggested in one of the above comments should make this feasible.

Reply:
I have amended the equivalent error message at the top to print the same thing as here.